### DISCUSSION TITLE
fevr-flask install

### DISCUSSION THREAD

**Discussion by renarena** [BeardedTek-com/fEVR#57] [Q&A]
I cloned this git repo, created .env and docker-compose (without tailscale), changed ip and mqtt user data.
What have i to put into `API_KEY`? It says to add key per api, but without a running fevr...

without the key or with an manually added long string i get
```
fevr_mqtt     | mqtt_client: error: the following arguments are required: key
fevr_mqtt     | usage: mqtt_client [-h] [-p PORT] [-t TOPICS] [-u USER] [-P PASSWORD]
fevr_mqtt     |                    [-f FEVR] [-s]
fevr_mqtt     |                    mqtt key
```

**Comment by renarena** [comment]
As soon as i created this question i am at least able to open the website.
It has an error (/setup/admin):
```
sqlalchemy.exc.OperationalError: (sqlite3.OperationalError) unable to open database file
(Background on this error at: https://sqlalche.me/e/14/e3q8)
```
have i to change any rights?

**Reply by BeardedTek** [reply]
Yes, by default the container is run as user 1000 (for now this isn't changeable)
in the folder which contains fEVR run:
```
sudo chown -R 1000 fevr
```

**Comment by BeardedTek** [comment]
To obtain your API Key:
- Click on the Menu Button:
![Menu](https://user-images.githubusercontent.com/93575915/165972395-74fbc3f1-1d59-4516-8eac-cc610b69f5e2.png)
- Click Edit Profile:
![EditProfile](https://user-images.githubusercontent.com/93575915/165972638-8352e3b0-e40d-4867-a494-c0d6c40fe7ad.png)
- Enter API Key Information:
![AddAPIKey](https://user-images.githubusercontent.com/93575915/165972877-1b6e8028-bc10-4087-b4d6-286168bc9638.png)
![APIKeyFilledIn](https://user-images.githubusercontent.com/93575915/165973165-6e5d8a3d-f87c-4c5a-9781-f22322d1ee26.png)
- Click Add Key and copy the API Key from this JSON Response:
![APIKeyResponse](https://user-images.githubusercontent.com/93575915/165973339-ac472a5c-330c-49de-94f7-58b5380d6d0f.png)

**Comment by renarena** [comment]
after executing `sudo chown -R 1000 fevr-flask `on the host it works :-)

on http://fevr-ip/setup/mqtt below the field api-key is an url shown 
The` : `between url / port is missing.

**Reply by BeardedTek** [reply]
Yeah, I'm trying to track down what's causing that.  I noticed it on one of my hosts.
You can edit run_mqtt_client.sh in the root where your docker-compose resides to correct that for now.

**Comment by renarena** [comment]
There is no run_mqtt_client.sh and i am not sure if i should use the fevr-flask repo or fevr.  
So i deleted fevr-flask and tried again with fevr.
Added admin user and frigate in settings.
On adding mqtt settings i got 
```
 Traceback (most recent call last):
fevr_flask    |   File "/fevr/venv/lib/python3.10/site-packages/flask/app.py", line 2095, in __call__
fevr_flask    |     return self.wsgi_app(environ, start_response)
fevr_flask    |   File "/fevr/venv/lib/python3.10/site-packages/flask/app.py", line 2080, in wsgi_app
fevr_flask    |     response = self.handle_exception(e)
fevr_flask    |   File "/fevr/venv/lib/python3.10/site-packages/flask/app.py", line 2077, in wsgi_app
fevr_flask    |     response = self.full_dispatch_request()
fevr_flask    |   File "/fevr/venv/lib/python3.10/site-packages/flask/app.py", line 1525, in full_dispatch_request
fevr_flask    |     rv = self.handle_user_exception(e)
fevr_flask    |   File "/fevr/venv/lib/python3.10/site-packages/flask/app.py", line 1523, in full_dispatch_request
fevr_flask    |     rv = self.dispatch_request()
fevr_flask    |   File "/fevr/venv/lib/python3.10/site-packages/flask/app.py", line 1509, in dispatch_request
fevr_flask    |     return self.ensure_sync(self.view_functions[rule.endpoint])(**req.view_args)
fevr_flask    |   File "/fevr/venv/lib/python3.10/site-packages/flask_login/utils.py", line 277, in decorated_view
fevr_flask    |     return current_app.ensure_sync(func)(*args, **kwargs)
fevr_flask    |   File "/fevr/app/setup.py", line 203, in setupAddMqttPost
fevr_flask    |     MQTT = mqtt(port=port,topics=topics,user=user,password=password,https=https,fevr=fevr,broker=broker,key=key)
fevr_flask    | UnboundLocalError: local variable 'user' referenced before assignment

```
Restarting docker and logout/ir doesnt change it.
No run_mqtt_client.sh either.

**Reply by BeardedTek** [reply]
> fevr_flask    |     MQTT = mqtt(port=port,topics=topics,user=user,password=password,https=https,fevr=fevr,broker=broker,key=key)
> fevr_flask    | UnboundLocalError: local variable 'user' referenced before assignment

This line tells me that you didn't read the on screen instructions.  In the user and password boxes, it says to enter none if there is no username or password.  This is an issue I will address within the next week or so.  This was put in as a workaround until I had time to hash it out.

Please repeat but enter "none" with no quotes for user and password if you have no password.

**Comment by BeardedTek** [comment]
Now that I've had a chance to look over the code, (Dad duties got in the way.  Science Fair, Birthdays, Award Ceremonies, etc...)
I should have a fix for that bug out momentarily.

**Comment by BeardedTek** [comment]
Also, the RC Code is now resides in the main branch of this repository.  All references to fevr-flask will be removed today.

Over the next week I will be diving into some cleanup issues I've found and release RC2.
